### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by dragos-rebegea**

### CODE REVIEW

File: tests/encodeCallData.test.ts [hunk 1]
```diff
@@ -31,15 +38,35 @@ describe('Encoding Functions', () => {
...
             const result = encodeCallData(endpointName, gasLimit, args);
-            const expected = `0x01${expectedLengthHex}${expectedNameHex}${expectedGasLimitHex}${expectedNumArgumentsHex}${expectedFirstArgumentLengthHex}${expectedFirstArgumentValueHex}${expectedSecondArgumentLengthHex}${expectedSecondArgumentValueHex}`;
+            const expected = `${expectedLengthHex}${expectedNameHex}${expectedGasLimitHex}${ArgumentsPresentProtocolMarker}${expectedNumArgumentsHex}${expectedFirstArgumentLengthHex}${expectedFirstArgumentValueHex}${expectedSecondArgumentLengthHex}${expectedSecondArgumentValueHex}`;
+            expect(result).toEqual(expected);
+        });
+    });
+
+    describe('encodeCallData Function', () => {
+        test('works and generate the same encodedData as on go codec', () => {
+            const expected = '0000000c656e64706f696e744e616d650000000002faf0800100000002000000012500000009737472696e67417267';
```
Comment (iulianpascalau) [inline tests/encodeCallData.test.ts]: this should be correct :+1:

**Review by iulianpascalau — APPROVED** [review]

File: helpers/encodeCallData.ts [hunk 2]
```diff
@@ -35,8 +35,11 @@ export function addGasLimit(callData: string, gasLimit: number): string {
 }
 
 export function addArgs(callData: string, args: any[]): string {
-    const numArgsHex = new BigNumber(args.length).toString(16).padStart(uint32ArgBytes, '0');
+    if (args.length == 0) {
+        return '00'
```
Comment (andreibancioiu) [inline helpers/encodeCallData.ts]: So, if there are no arguments, there is no presence marker, and the number of arguments is 0. Sounds good.

File: helpers/encodeCallData.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { numberToPaddedHex } from "./utils.codec";
 
 const uint32ArgBytes = 8
 const uint64ArgBytes = 16
-const DataPresentProtocolMarker = "0x01"
+export const ArgumentsPresentProtocolMarker = "01"
```
Comment (andreibancioiu) [inline helpers/encodeCallData.ts]: From the name, it's not obvious that it's a hex value as string, but all right.

**Review by andreibancioiu — DISMISSED** [review]

File: helpers/encodeCallData.ts (hunk 2 above)
Comment (iulianpascalau) [reply helpers/encodeCallData.ts]: actually 00 means "no data presence marker"

Comment (iulianpascalau) [reply helpers/encodeCallData.ts]: a const would have made it clearer :)

File: helpers/encodeCallData.ts (hunk 3 above)
Comment (dragos-rebegea) [reply helpers/encodeCallData.ts]: renamed

File: helpers/encodeCallData.ts (hunk 2 above)
Comment (dragos-rebegea) [reply helpers/encodeCallData.ts]: added const

**Review by iulianpascalau — DISMISSED** [review]

**Review by andreibancioiu — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**